src/Disks/IO/WriteBufferFromAzureBlobStorage.cpp

```diff
@@ -173,24 +178,50 @@ void WriteBufferFromAzureBlobStorage::preFinalize()
             auto part_data = std::move(detached_part_data.front());
             Azure::Core::IO::MemoryBodyStream memory_stream(
                 reinterpret_cast<const uint8_t *>(part_data.memory.data()), part_data.data_size);
-            execWithRetry(
-                [&](size_t retry_attempt)
-                {
-                    Azure::Storage::Blobs::UploadBlockBlobOptions options;
 
-                    if (write_settings.object_storage_write_if_none_match.empty())
-                        options.AccessConditions.IfNoneMatch = Azure::ETag(write_settings.object_storage_write_if_none_match);
-
-                    if (write_settings.object_storage_write_if_match.empty())
-                        options.AccessConditions.IfMatch = Azure::ETag(write_settings.object_storage_write_if_match);
+            Stopwatch watch;
+            Int32 error_code = 0;
+            String error_message;
+            try
+            {
+                execWithRetry(
+                    [&](size_t retry_attempt)
+                    {
+                        Azure::Storage::Blobs::UploadBlockBlobOptions options;
+
+                        if (write_settings.object_storage_write_if_none_match.empty())
+                            options.AccessConditions.IfNoneMatch = Azure::ETag(write_settings.object_storage_write_if_none_match);
+
+                        if (write_settings.object_storage_write_if_match.empty())
+                            options.AccessConditions.IfMatch = Azure::ETag(write_settings.object_storage_write_if_match);
```

Comment:
btw looks like a bug here, it should be 
```suggestion
                        if (!write_settings.object_storage_write_if_none_match.empty())
                            options.AccessConditions.IfNoneMatch = Azure::ETag(write_settings.object_storage_write_if_none_match);

                        if (!write_settings.object_storage_write_if_match.empty())
                            options.AccessConditions.IfMatch = Azure::ETag(write_settings.object_storage_write_if_match);
```
it is the same in the old diff, so was always broken